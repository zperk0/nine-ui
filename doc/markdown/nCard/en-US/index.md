
# Card 

Display the information in the card container.

## When To Use

Use when you need to introduce something.

##  Examples

### Basic Use

:::demo
```html
<template>
  <n-card :src="src" :description="description" title="Bird"/>
</template>
<script>
  export default {
    data () {
      return {
        src: 'https://timgsa.baidu.com/timg?image&quality=80&size=b9999_10000&sec=1585736912147&di=b569099e49bb8d6216841e4819d98222&imgtype=0&src=http%3A%2F%2Fa4.att.hudong.com%2F21%2F09%2F01200000026352136359091694357.jpg',
        description:'Birds, also called birds. Definition: Oviparous vertebrates covered with feathers on the body surface. The main characteristics of birds are: the body is streamlined (spindle or spindle), and most of them fly.'
      }
    }
  }
</script>
```
:::

## API

| Property | Description | Type | 
| :--- | :--- | :--- | :--- |
| src | Picture address | String | 
| title | Card title | String |  
| description | Content introduction | String |  



