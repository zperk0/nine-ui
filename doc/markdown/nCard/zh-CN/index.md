
# Card 

将信息在卡片容器中展示。

## 何时使用

需要介绍某种事物时使用。

## 代码演示

### 基本使用

:::demo
```html
<template>
  <n-card :src="src" :description="description" title="鸟"/>
</template>
<script>
  export default {
    data () {
      return {
        src: 'https://timgsa.baidu.com/timg?image&quality=80&size=b9999_10000&sec=1585736912147&di=b569099e49bb8d6216841e4819d98222&imgtype=0&src=http%3A%2F%2Fa4.att.hudong.com%2F21%2F09%2F01200000026352136359091694357.jpg',
        description:'鸟，又称作鸟儿。定义：体表被覆羽毛的卵生脊椎动物，鸟的主要特征是：身体呈流线型（纺锤型或梭形），大多数飞翔生活。'
      }
    }
  }
</script>
```
:::

## API

| 参数 | 说明 | 类型 | 
| :--- | :--- | :--- | :--- |
| src | 图片地址 | String | 
| title | 卡片标题 | String |  
| description | 介绍内容 | String |  



