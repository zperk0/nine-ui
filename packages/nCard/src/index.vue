<template>
  <div class="card">
    <img
      class="card-img"
      :src="src"
    >
    <span class="card-title">{{ title }}</span>
    <p class="card-message">
      {{ description }}
    </p>
  </div>
</template>

<script>
export default {
  name: 'NCard',
  props: {
    title: {
      type: String,
      default: ''
    },
    src: {
      type: String,
      default: ''
    },
    description: {
      type: String,
      default: ''
    }
  }
};
</script>

<style lang="scss" scoped>
@import './index.scss'
</style>
