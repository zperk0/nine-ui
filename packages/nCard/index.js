import nCard from './src/index.vue';

nCard.install = function (Vue) {
  Vue.component(nCard.name, nCard);
};

export default nCard;
