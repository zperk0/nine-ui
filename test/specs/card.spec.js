import nCard from '../../packages/n-card';
import { mount } from '@vue/test-utils';

describe('nCard', () => {
  it('should return nCard', () => {
    const wrapper = mount({
      render () {
        return <nCard />;
      }
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
  it('should render correct title', () => {
    const wrapper = mount({
      render () {
        return <nCard title="test"/>;
      }
    });
    expect(wrapper.text()).toBe('test');
  });
  it('should render correct description', () => {
    const wrapper = mount({
      render () {
        return <nCard description="description"/>;
      }
    });
    expect(wrapper.text()).toBe('description');
  });
});
