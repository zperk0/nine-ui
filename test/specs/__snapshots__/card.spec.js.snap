// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`nCard should return nCard 1`] = `
<div class="card"><img class="card-img"><span class="card-title"><!----></span>
  <p class="card-message">
    <!---->
  </p>
</div>
`;
